lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,458 @@
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.CreateAdIntegration = enableAdIntegration
+	}
+}
+
+// WithAdApplicationId Set Active Directory application id
+func WithAdApplicationId(AdApplicationId string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AdApplicationId = AdApplicationId
+	}
+}
+
+// WithAdApplicationPassword Set the Active Directory password
+func WithAdApplicationPassword(AdApplicationPassword string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AdApplicationPassword = AdApplicationPassword
+	}
+}
+
+// WithAdServicePrincipalId Set Active Directory principal id
+func WithAdServicePrincipalId(AdServicePrincipalId string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AdServicePrincipalId = AdServicePrincipalId
+	}
+}
+
+// WithManagementGroup Enable the Management Group to allow AD to be reader on management group
+// rather then subscription
+func WithManagementGroup(enableManagentGroup bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ManagementGroup = enableManagentGroup
+	}
+}
+
+// WithManagementGroupId The Group Id to add reader permissions
+func WithManagementGroupId(managementGroupId string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ManagementGroupId = managementGroupId
+	}
+}
+
+// WithSubscriptionIds List of subscriptions to to enable logging
+func WithSubscriptionIds(subscriptionIds []string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.SubscriptionIds = subscriptionIds
+	}
+}
+
+// WithAllSubscriptions Grant read access to ALL subscriptions within
+// the selected Tenant (overrides 'subscription_ids')
+func WithAllSubscriptions(allSubscriptions bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AllSubscriptions = allSubscriptions
+	}
+}
+
+// WithExistingStorageAccount Use an existing Storage Account
+func WithExistingStorageAccount(existingStorageAccount bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ExistingStorageAccount = existingStorageAccount
+	}
+}
+
+// WithStorageAccountName The name of the Storage Account
+func WithStorageAccountName(storageAccountName string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.StorageAccountName = storageAccountName
+	}
+}
+
+// WithStorageAccountResourceGroup The Resource Group for the existing Storage Account
+func WithStorageAccountResourceGroup(storageAccountResourceGroup string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.StorageAccountResourceGroup = storageAccountResourceGroup
+	}
+}
+
+// WithLocation The Azure region where storage account for logging is
+func WithLocation(location string) AzureTerraformModifier {
```

Comment:
IMO we should rename this to `WithStorageAccountLocation` seeing as this location is only applicable to the Storage Account

Reply:
Yeah, renamed the attribute to `StorageLocation` to make it clearer